File: 01gpio/062dth/dht11.js
var sensor = require('node-dht-sensor');
var dht = 11;
var pin = 18;

function dht_cb(err, temperature, humidity) {
    if (!err) {
        console.log('temp: ' + temperature.toFixed(1) + '°C, ' +
            'humidity: ' + humidity.toFixed(1) + '%'
        );
    }else{
    	console.log(err);
    }
    setTimeout(function(){
    	sensor.read(dht, pin, dht_cb);
    }, 100);
    
}
sensor.read(dht, pin, dht_cb);

File: 04web_iot/dht11.py
from flask import *
import RPi.GPIO as GPIO
import Adafruit_DHT, json, time

app = Flask(__name__)

GPIO.setmode(GPIO.BCM)
sensor = Adafruit_DHT.DHT11
pin_dht11 = 24

@app.route('/')
def main():
    #return app.send_static_file('dht11.html')
    return app.send_static_file('dht11_gauge.html')

@app.route('/monitor')
def monitoring():
    try:
        humidity, temperature = Adafruit_DHT.read_retry(sensor, pin_dht11)
        obj = {'humi' : humidity, 'temp' : temperature}
        return Response(json.dumps(obj), content_type='application/json')
    except Exception as e:
        print('err', e)


if __name__ == '__main__':
    try:
        app.run(host='0.0.0.0', port=8080, debug=True)
    finally:
        GPIO.cleanup()
        print('cleaning up')